applications/portfolioDrafts/getPortfolioDrafts.ts

```diff
@@ -0,0 +1,70 @@
+import { ScanCommand, ScanCommandInput } from "@aws-sdk/lib-dynamodb";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { ErrorCodes } from "./models/Error";
+import { PortfolioDraftSummary } from "./models/PortfolioDraftSummary";
+import { dynamodbDocumentClient as client } from "./utils/dynamodb";
+import { ApiSuccessResponse, ErrorResponse, ErrorStatusCode, SuccessStatusCode } from "./utils/response";
+
+const TABLE_NAME = process.env.ATAT_TABLE_NAME;
+const QUERY_PARAM_INVALID = new ErrorResponse(
+  { code: ErrorCodes.INVALID_INPUT, message: "Invalid request parameter" },
+  ErrorStatusCode.BAD_REQUEST
+);
+
+/**
+ * Convert string to integer, return defaultInt if undefined or NaN.
+ * @param str - The string value to convert
+ * @param defaultInt - The default integer value to return if str undefined or NaN
+ */
+function getIntegerOrDefault(str: string | undefined, defaultInt: number): number {
+  if (str === undefined) return defaultInt;
+  const int = parseInt(str);
+  if (isNaN(int)) return defaultInt;
+  return int;
+}
```

Comment:
This may not necessarily do what we want. It seems like this is being used primary to evaluate query string parameters. The issue here is that users pass bad data all the time and `parseInt` doesn't actually fail on non-integer inputs. For example `parseInt("123abc") == 123`. We also would accept an offset of `234.123` as being `234`. What we _may_ want to do is instead:

```suggestion
function getIntegerOrDefault(str?: string, defaultInt: number): number {
  // ?. should protect against undefined and null while asserting the input value is numeric
  if (!str?.match(/$[0-9]+^/)) {
    return defaultInt;
  }
  // NaN should not be a possible for `parseInt` with a numeric input.
  return parseInt(str);
}
```

Reply:
Attempting to combine this advice with @jeffsegal-ccpo observations about default params.

Reply:
I like the inclusion of the regex.  Smart.
Caret (^) marks the beginning of the string and dollar sign ($) marks the end, however.

Attempting to change the type for `str` as prescribed results in _A required parameter cannot follow an optional parameter.ts(1016)_ so will retrain `string | undefined`.

Reply:
I've incorporated this suggestion with some minor slight changes.